onnx/shape_inference/implementation.cc

```diff
@@ -388,6 +393,8 @@ class ShapeInferenceImplBase {
           updateType(n.output(i), ctx.getOutputType(i));
       }
 
+      preprocess(n);
```

Comment:
IIUC, `preprocess` is being called twice now, and we should drop one of them.

I suggest dropping the loop in line 474-476 and in 540-542, since you have it above as part of `process(node)`.

But I would move the above line to the beginning of this function, before type-and-shape-inference is called.

Reply:
Or, drop the above line, and retain the loops in 474 and 540.